core/src/main/java/com/yahoo/oak/EntryHashSet.java

```diff
@@ -240,13 +240,26 @@ private EntryState getEntryState(ThreadContext ctx, int idx, K key, int keyHash)
             return EntryState.VALID;
         }
 
-        // value is invalid
+        // value is invalid, in case another thread inserts the same key
         if (isKeyAndEntryKeyEqual(ctx.key, key, idx, keyHash)) {
             return EntryState.INSERT_NOT_FINALIZED;
         }
         return EntryState.VALID;
     }
 
+    /**
+     * Function checks if an entry at given index is valid.
```

Comment:
Entry at given index is valid "from scan perspective". It is maybe "valid enough" for scan, but not for some other operation... Please think about more precise name and function explanation.

Reply:
if the readValue succeeds, the entry is valid from any perspective, isn't it?

Reply:
If there is no concurrent deletion -- yes. Otherwise deletion of this entry can already been started.